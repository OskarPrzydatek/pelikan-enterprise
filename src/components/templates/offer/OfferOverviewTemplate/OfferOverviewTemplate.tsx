import { OverviewList } from '~/components/organisms';
import { PageLayout } from '~/styles';

interface IOfferOverviewTemplate {
  page?: string;
  onClickNavigate: () => void;
}

export const OfferOverviewTemplate: React.FC<IOfferOverviewTemplate> = ({
  page,
  onClickNavigate,
}: IOfferOverviewTemplate) => {
  return (
    <PageLayout>
      <OverviewList
        navigateLabel="Dodaj ofertÄ™"
        noItemsLabel="Brak ofert w systemie"
        page={page}
        title="Oferty"
        onClickNavigate={onClickNavigate}
      />
    </PageLayout>
  );
};
