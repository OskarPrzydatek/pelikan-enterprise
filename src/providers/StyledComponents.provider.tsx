import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
