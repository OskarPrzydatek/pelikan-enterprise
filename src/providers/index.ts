export { StyledComponentsProvider } from './StyledComponents.provider';
