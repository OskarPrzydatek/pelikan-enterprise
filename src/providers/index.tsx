import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

interface ITestingPageProvider {
  element: JSX.Element;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};

export const TestingPageProvider: React.FC<ITestingPageProvider> = ({
  element,
}: ITestingPageProvider) => {
  return (
    <StyledComponentsProvider>
      <MemoryRouter initialEntries={['/page1']}>
        <Routes>
          <Route element={element} path="/:page" />
        </Routes>
      </MemoryRouter>
    </StyledComponentsProvider>
  );
};
