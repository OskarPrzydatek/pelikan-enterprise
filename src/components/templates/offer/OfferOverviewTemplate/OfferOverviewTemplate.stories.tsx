import { Meta, StoryObj } from '@storybook/react';

import { StyledComponentsProvider } from '~/providers';

import { OfferOverviewTemplate } from './OfferOverviewTemplate';

const meta: Meta<typeof OfferOverviewTemplate> = {
  title: 'Templates/OfferOverviewTemplate',
  component: OfferOverviewTemplate,
};

export default meta;
type Story = StoryObj<typeof OfferOverviewTemplate>;

export const OfferOverviewTemplatePrimmary: Story = {
  render: () => (
    <StyledComponentsProvider>
      <OfferOverviewTemplate
        onClickNavigate={() => console.log('onClickNavigate')}
      />
    </StyledComponentsProvider>
  ),
};
