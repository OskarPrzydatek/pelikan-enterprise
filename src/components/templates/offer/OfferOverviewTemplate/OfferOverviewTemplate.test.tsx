import { render } from '@testing-library/react';
import { describe, test, vi } from 'vitest';

import { StyledComponentsProvider } from '~/providers';

import { OfferOverviewTemplate } from './OfferOverviewTemplate';

const mockOnClickNavigate = vi.fn();

describe('OfferOverviewTemplate', () => {
  test('component snapshot', () => {
    const view = render(
      <StyledComponentsProvider>
        <OfferOverviewTemplate onClickNavigate={mockOnClickNavigate} />
      </StyledComponentsProvider>
    );
    expect(view).toMatchSnapshot();
  });
});
