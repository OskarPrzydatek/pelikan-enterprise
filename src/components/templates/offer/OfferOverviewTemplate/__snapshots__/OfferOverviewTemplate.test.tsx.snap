// Vitest Snapshot v1

exports[`OfferOverviewTemplate > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gswNZR cJrdyV"
      >
        <div
          class="sc-gYbzsP fZAhDs"
        >
          <h2
            class="sc-eDvSVe dcXtK"
          >
            Oferty
          </h2>
          <ul
            class="sc-cCjUiG gdYusC"
          >
            <li>
              <p
                class="sc-ftTHYK iSkTh"
              >
                Brak ofert w systemie
              </p>
            </li>
          </ul>
          <div
            class="sc-cjibBx bzGlRW"
          >
            <p
              class="sc-ftTHYK iSkTh"
              data-testid="pagination-counter"
            >
              1
               / X
            </p>
          </div>
          <button
            class="sc-hLBbgP fxgbmT"
          >
            Dodaj ofertę
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gswNZR cJrdyV"
    >
      <div
        class="sc-gYbzsP fZAhDs"
      >
        <h2
          class="sc-eDvSVe dcXtK"
        >
          Oferty
        </h2>
        <ul
          class="sc-cCjUiG gdYusC"
        >
          <li>
            <p
              class="sc-ftTHYK iSkTh"
            >
              Brak ofert w systemie
            </p>
          </li>
        </ul>
        <div
          class="sc-cjibBx bzGlRW"
        >
          <p
            class="sc-ftTHYK iSkTh"
            data-testid="pagination-counter"
          >
            1
             / X
          </p>
        </div>
        <button
          class="sc-hLBbgP fxgbmT"
        >
          Dodaj ofertę
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
